db_test.go

```diff
@@ -1309,6 +1309,112 @@ func TestInitializeHeadTimestamp(t *testing.T) {
 	})
 }
 
+func TestNoEmptyBlocks(t *testing.T) {
+	db, close := openTestDB(t, nil)
+	defer close()
+	defer db.Close()
+	db.DisableCompactions()
+
+	// Test no blocks after compact with empty head.
+	testutil.Ok(t, db.compact())
+	bb, err := blockDirs(db.Dir())
+	testutil.Ok(t, err)
+	testutil.Equals(t, len(db.Blocks()), len(bb))
+	testutil.Equals(t, 0, len(bb))
+
+	// Test no blocks after deleting all samples from head.
+	blockRange := DefaultOptions.BlockRanges[0]
+	label := labels.FromStrings("foo", "bar")
```

Comment:
can you rename this to `defaultLabel` and use it everywhere as `labels.NewEqualMatcher(defaultLabel[0].Name, defaultLabel[0].Value))